### ISSUE TITLE
[Workflow] Allow retrieving transitions by name.

### ISSUE THREAD

**Issue by d-ph** [symfony/symfony#24480]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.10

Hello,

Would it be possible to add a way to retrieve transitions by name from a Workflow, please? Something like the following: `Workflow::getTransitionByName(string): ?Transition` and `Workflow::getTransitions(): array`.

Motivation: I have a generic code for retrieving workflows for different objects and transitioning them using different transition names. I'd like to throw an exception from my generic code, when the caller tries to transition an object using an undefined transition name (missing workflow transition case). Currently, there's no way to retrieve all defined transitions in a workflow. `Workflow::getEnabledTransitions(): array` returns possible transitions in current object's state only, which doesn't solve my problem.

Thank you,
@d-ph

**Comment by lyrixx** [comment]
Hello @d-ph 

Thanks for your interest in the Workflow component.

It's not easy to discuss two topics in the same Github issue.
Could you move the second part of this issue in another/new issue?

(So here, I will only comment the first topic)

Note: I will delete this comment once you moved the second topic

**Comment by lyrixx** [comment]
> Currently, there's no way to retrieve all defined transitions in a workflow

You can do it with the following code: 

```php
$workflow->getDefinition()->getTransitions();
```

About: `Workflow::getTransitionByName(string): ?Transition` I think you can create it in your code based on `$workflow->getDefinition()->getTransitions();`

Note: If you are using the full stack framework and a `type: state_machine` workflow, you can have many transition with the same name. See [this name](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L506-L512). And if you are building the workflow by yourself, you can also have many transitions with the same name.

Is it enough for you?

**Comment by d-ph** [comment]
Hello @lyrixx 

Thanks for your reply.

You're right about having two topic in the same Github issue. I moved the second one to a separate ticket.

And you're right about the ability to retrieve the transitions via the `::getDefinition()` method. I kind of feel embarrassed now, because I could've sworn I checked for the presence of this getter and couldn't find it. But it's there. And yes, I'm perfectly fine to retrieve transitions by their names in my code using the method you suggested. And thanks for mentioning, that Workflow allows to have transitions with the same names -- I wasn't aware of that.

This pretty much closes this ticket. Again, thanks for your quick answer and the suggestion on how I can go about achieving what I need to.
